neptune/projects.py

```diff
@@ -412,14 +414,25 @@ def create_experiment(self,
             if os.path.isfile(main_file):
                 entrypoint = normalize_file_name(os.path.basename(main_file))
                 upload_source_entries = [
-                    UploadEntry(os.path.abspath(main_file), normalize_file_name(os.path.basename(main_file)))
+                    UploadEntry(abspath(main_file), normalize_file_name(os.path.basename(main_file)))
                 ]
         else:
             expanded_source_files = set()
             for filepath in upload_source_files:
                 expanded_source_files |= set(glob(filepath))
+            absolute_paths = set()
             for filepath in expanded_source_files:
-                upload_source_entries.append(UploadEntry(os.path.abspath(filepath), normalize_file_name(filepath)))
+                absolute_paths.add(abspath(filepath))
+            try:
+                common_source_root = commonpath(absolute_paths)
+                if os.path.isfile(common_source_root):
+                    common_source_root = os.path.dirname(common_source_root)
+                for absolute_path in absolute_paths:
+                    upload_source_entries.append(UploadEntry(absolute_path, normalize_file_name(os.path.relpath(absolute_path, common_source_root))))
+            except ValueError:
+                for absolute_path in absolute_paths:
+                    upload_source_entries.append(UploadEntry(absolute_path, normalize_file_name(absolute_path)))
```

Comment:
1. When error occurs on last file you re-upload them all
2.Where exactly does the ValueError come from?

Reply:
This only collects UploadEntry in the memory, does not send files anywhere. Actual sending is done later, out of scope of this change.